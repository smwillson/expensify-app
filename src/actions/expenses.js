import uuid from 'uuid';

//ADD_EXPENSE --> returns an object by default

export const addExpense = ({
    desc = '',
    note = '',
    amount = 0,
    createdAt = 0 } = {}
) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        desc,
        note,
        amount,
        createdAt
    }
});

//REMOVE_EXPENSE --> returns an object by default
export const removeExpense = ({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
});


//EDIT_EXPENSE --> returns an object by implicitly

export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates

});