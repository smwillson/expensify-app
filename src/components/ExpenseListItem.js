import React from 'react';
import { Link } from "react-router-dom";

const ExpenseListItem = ({id, desc, amount, createdAt }) => (
    <div>
        <Link to={`/edit/${id}`}> <h3>{desc}</h3></Link>
        <div> {amount} {createdAt}</div>
    </div>
);

export default ExpenseListItem;