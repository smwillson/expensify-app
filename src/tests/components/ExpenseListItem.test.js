import React from 'react';
import expenses from '../fixtures/expenses';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';

test('should render an expense list item',()=>{
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();

});