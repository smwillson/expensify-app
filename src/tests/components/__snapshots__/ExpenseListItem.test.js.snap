// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an expense list item 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    <h3>
      rent
    </h3>
  </Link>
  <p>
    $195.00
    -
    December 31st, 1969
  </p>
</div>
`;
